sdk/metric/view/view.go

```diff
@@ -17,22 +17,195 @@
 
 package view // import "go.opentelemetry.io/otel/sdk/metric/view"
 
-// Config contains configuration options for a view.
-type Config struct {
-	// TODO (#2837): implement.
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/sdk/instrumentation"
+)
+
+// View provides users with the flexibility to customize the metrics that are
+// output by the SDK. A View can be used to:
+//
+// * Ignore Instruments.
+// * Change the name of an Instrument.
+// * Change the aggregation of an Instrument.
+// * Customize which attribute(s) are to be reported by the Instrument.
+//
+// An empty config will match all instruments, and do no modifications.
+type View struct {
+	instrumentName *regexp.Regexp
+	hasWildcard    bool
+	scope          instrumentation.Library
+
+	filterKeys  map[attribute.Key]struct{}
```

Comment:
Should this be stored as an `attribute.Filter` instead?

Reply:
Currently the Attributes will be additive, eg
```go
WithAttributeFilter("foo"),
WithAttributeFilter("bar"),
```
will keep both attributes.  If we change this behavior to replace, then I think it makes more sense to store the filter.  We are not very uniform across the SDK on when something like this should be additive or should replace the last value.

Reply:
Gotcha. Can we make it clear in the `WithAttributeFilter` docs that it is additive if provided multiple times.

Reply:
I added the comment at the comment at the top of New

> If there are any duplicate Options passed, the last one passed will take precedence.

And I made the option a replacement option.  So we are now storing the filter.  Does that work?

Reply:
LGTM :+1: